tools/autograd/gen_variable_type.py

```diff
@@ -48,8 +48,22 @@
 
 # These functions have their names recorded under trace renamed,
 RENAME_TRACE = {
-    'zero': 'zeros_like',
-    'fill': 'full_like',
+    'zero': 'zeros_like',  # replacing aten::zero_ with aten::zeros_like
+    'fill': 'full_like',  # replacing aten::fill_ with aten::full_like
+}
+
+# In some cases outplace replacement of the operator requires more arguments.
+# We add them manually as we don't want to add unused arguments into the
+# native_functions.yaml
```

Comment:
Hmm, I don't really understand. Could you explain in more detail? What is "outplace replacement"? What are the "unused arguments"?

Reply:
I think a before/after description would help.

Reply:
I will copy this into the code, so it is searchable by the ops name.

torch.jit.trace have undocumented keyword argument _force_outplace, which force jit to replace functions with outplace variants (for example aten::add_ becomes aten::add).
This replacement implemented in-place with minimum modifications of arguments stack (as it assumes that outplace will have to save arguments as inplace version).
However there are no such substitutions available for aten::fill_ and aten::zero_ operators, as we sever implemented aten::fill and aten::zero, so jit tracing hack replacing aten::zero_ with aten::zeros_like and replacing aten::fill_ with aten::full_like. But as they potentially can have different arguments, we also have to hack into the stack and add missing ones.

Reply:
OK makes sense. I half wonder if it wouldn't have been cleaner to make variants of `full_like` and `zeros_like` which don't take extra factory arguments, so the convention lines up exactly, but this is quibbling.